src/na/zoned/decomposer/NativeGateDecomposer.cpp

```
@@ -0,0 +1,275 @@
+    SingleQubitGateLayer MidLayer;
+    SingleQubitGateLayer BackLayer;
+    SingleQubitGateLayer NewLayer;
+
+    for (auto gate : layer) {
+      std::array<qc::fp, 3> decomp_angles =
+          getDecompositionAngles(gate.angles, theta_max);
+
+      // GR(theta_max/2, PI_2)==Global Y due to PI_2
+      auto sop = qc::StandardOperation(gate.qubit, qc::RZ, {decomp_angles[1]});
+      std::unique_ptr<const qc::Operation> op =
+          std::make_unique<const qc::StandardOperation>(sop);
```

Cppcheck (2.19.0)

[style] The function 'fromJSONString' is never used.

(unusedFunction)